src/GraphQL/Value.hs

```diff
@@ -151,6 +161,11 @@ toObject _ = empty
 -- * Scalars
 
 -- | A non-variable value which contains no other values.
+--
+-- Note that the 'FromJSON' instance always decodes JSON strings to
+-- GraphQL strings (never enums) and JSON numbers to GraphQL floats
+-- (never ints); doing a better job of resolving this requires query
+-- context.
```

Comment:
I'm a little uncomfortable about this. The "Zen of Python" advises "in the face of ambiguity, refuse the temptation to guess", and I think that's good advice. 

Some options I could think of:
- _don't_ have a `FromJSON` instance for this, and instead have a series of functions that take the query (or whatever the minimal necessary context is) and a `Value` to be parsed, so that we can unambiguously parse these things
- create a new type (types?) that unites the query context with these values and write `FromJSON` instances for these new types
- create a new `AmbiguousConstScalar` type. Then, have a function that transforms `AmbiguousConstScalar -> ConstScalar` by also taking a query context. This type would need fewer branches than `ConstScalar`.
- make `ConstScalar` a phantom type, where the phantom parameter is whether it's ambiguous or not. Then, have a function that transforms `ConstScalar Ambiguous -> ConstScalar Unambiguous` by also taking a query context